presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java

```diff
@@ -551,7 +552,34 @@ private PlanBuilder aggregate(PlanBuilder subPlan, QuerySpecification node)
         if (needPostProjectionCoercion) {
             return explicitCoercionFields(subPlan, distinctGroupingColumns, analysis.getAggregates(node));
         }
-        return subPlan;
+
+        // 4. Project and re-write all grouping functions
+        return handleGroupingOperations(subPlan, node, groupIdSymbol);
+    }
+
+    private PlanBuilder handleGroupingOperations(PlanBuilder subPlan, QuerySpecification node, Optional<Symbol> groupIdSymbol)
+    {
+        if (analysis.getGroupingOperations(node).isEmpty()) {
+            return subPlan;
+        }
+
+        TranslationMap translations = subPlan.copyTranslations();
+
+        Assignments.Builder projections = Assignments.builder();
+        projections.putIdentities(subPlan.getRoot().getOutputSymbols());
+
+        for (GroupingOperation groupingOperation : analysis.getGroupingOperations(node)) {
+            Expression rewritten = GroupingOperationRewriter.rewriteGroupingOperation(groupingOperation, node, analysis, metadata, groupIdSymbol);
+            // Do not project a rewritten grouping operation that has already been projected
```

Comment:
It has been projected, but on the same level (e.g: the projection is not yet added), so you cannot reuse the symbol yet. I suppose this might fail:
```
select grouping(a), grouping(a) from ... group by a
```
Add test for that